src/helper/worker_pool.hpp

```diff
@@ -45,37 +142,21 @@ class monitor {
 class pool {
 public:
     pool() = default;
-    ~pool()
-    {
-        if (wm_.running()) {
-            stop();
-        }
-    }
-
+    ~pool() = default;
     pool(const pool &) = delete;
     pool &operator=(const pool &) = delete;
     pool(pool &&) = delete;
     pool &operator=(pool &&) = delete;
 
-    template <class Function, class... Args>
-    bool launch(Function &&f, Args &&...args)
-    {
-        if (!wm_.running()) {
-            return false;
-        }
-        std::thread t(std::forward<Function>(f), std::ref(wm_),
-            std::forward<Args>(args)...);
-        t.detach();
-        return true;
-    }
+    bool launch(runnable &&f);
 
-    [[nodiscard]] unsigned worker_count() const { return wm_.count(); }
+    void wait() { q_.wait(); }
```

Comment:
producer::queue::stop() already waits

Reply:
Calling wait on its own allows us to finalise processing before exiting, it's not used for now...